src/rdkafka_request.c

```diff
@@ -476,93 +476,177 @@ rd_kafka_FindCoordinatorRequest(rd_kafka_broker_t *rkb,
+                return NULL;
+        }
+        total_partitions = topic_partitions->cnt;
+        rkbuf = rd_kafka_buf_new_request(rkb, RD_KAFKAP_ListOffsets, 1,
+                                         (total_partitions * 40) + 20);
+        /* Replica ID */
+        rd_kafka_buf_write_i32(rkbuf,-1);
+        if(ApiVersion > 1)
+                rd_kafka_buf_write_i8(rkbuf,isolation_level);
+                 
+        
+        of_topic = rd_kafka_buf_write_i32(rkbuf,0);;
+        for(i=0;i<total_partitions;i++){
+                rd_kafka_topic_partition_t *topic_partition = &topic_partitions->elems[i];
+                if(strcmp(topic_partition->topic,prevtopic)!=0){
+                        /* Write partition cnt of previous topic */
+                        if(num_topics){
+                                rd_kafka_buf_update_i32(rkbuf,of_partition,num_partitions);
+                                if(ApiVersion >= 6)
+                                        rd_kafka_buf_write_tags(rkbuf);
                         }
+                        num_topics++;
+                        num_partitions = 1;
+                        /* Write the topic */
+                        rd_kafka_buf_write_str(rkbuf,topic_partition->topic,-1);
+                        of_partition = rd_kafka_buf_write_i32(rkbuf,0);
+                        prevtopic = topic_partition->topic;
+                }else{
+                        num_partitions++;
+                }
 
-                        rktpar = rd_kafka_topic_partition_list_add(
-                            offsets, topic_name, kpartition);
-                        rktpar->err    = ErrorCode;
-                        rktpar->offset = Offset;
-                        rd_kafka_topic_partition_set_leader_epoch(rktpar,
-                                                                  LeaderEpoch);
+                /* Partition Index*/
+                rd_kafka_buf_write_i32(rkbuf,topic_partition->partition);
 
-                        if (ErrorCode && !all_err)
-                                all_err = ErrorCode;
-                }
-        }
+                /* Current Leader Epoch Version 4 Onwards */
+                if(ApiVersion >= 4)
+                        rd_kafka_buf_write_i32(rkbuf,0);
 
-        return all_err;
+                /* Timestamp */
+                rd_kafka_buf_write_i64(rkbuf,topic_partition->offset);
 
-err_parse:
-        return rkbuf->rkbuf_err;
-}
+                /* Max_Num_Offsets in version 0*/
+                if(ApiVersion == 0)
+                        rd_kafka_buf_write_i32(rkbuf,1);
 
+                if(ApiVersion >= 6)
+                        rd_kafka_buf_write_tags(rkbuf);
+        }
+        if(num_topics){
+                rd_kafka_buf_update_i32(rkbuf,of_partition,num_partitions);
+                rd_kafka_buf_update_i32(rkbuf,of_topic,num_topics);
+        }
+        if(ApiVersion >= 6)
+                rd_kafka_buf_write_tags(rkbuf);
```

Comment:
Not needed for top level buffer, as it's written by `rd_kafka_buf_finalize`.
Please remove this final write_tags for `rd_kafka_DescribeUserScramCredentialsRequest` and `rd_kafka_AlterUserScramCredentialsRequest` too. Last skip_tags is also not strictly necessary as it's possible to partially read the buffer.